spec/features/applications/create_spec.rb

```diff
@@ -63,5 +63,41 @@
       expect(page).to have_content("In Progress")
     end
 
+    it "if form is not filled in, will display an error and re-render the form" do
+      visit "/applications/new"
+      click_on "Submit"
+
+      expect(page).to have_current_path("/applications/new")
+      expect(page).to have_content("Error: Name can't be blank, Street can't be blank, City can't be blank, State can't be blank, Zip can't be blank, Description can't be blank, Zip is the wrong length (should be 5 characters), State Please use 2 capital letters for the state abbreviation")
+    end
+
+    it "if form is missing a single field, will display an error and re-render the form" do
+      visit "/applications/new"
+
+      fill_in "Name", with: "Jill"
+      fill_in "Street", with: "99 Memory Ln."
+      fill_in "State", with: "UT"
+      fill_in "Zip", with: "12345"
+      fill_in "Description", with: "The liger is my fav animal"
+      click_on "Submit"
+
+      expect(page).to have_current_path("/applications/new")
+      expect(page).to have_content("Error: City can't be blank")
+    end
+
+    it "if :state ro :zip are incorrecly formatted, will display an error and re-render the form" do
+      visit "/applications/new"
+
+      fill_in "Name", with: "Jill"
+      fill_in "Street", with: "99 Memory Ln."
+      fill_in "city", with: "Townsville"
+      fill_in "State", with: "az"
+      fill_in "Zip", with: "1234"
+      fill_in "Description", with: "The liger is my fav animal"
+      click_on "Submit"
+
+      expect(page).to have_current_path("/applications/new")
+      expect(page).to have_content("Error: Zip is the wrong length (should be 5 characters), State Please use 2 capital letters for the state abbreviation")
```

Comment:
```suggestion
      expect(page).to have_content("Error: Zip is the wrong length (should be 5 characters), State must be an abbreviation of two capital letters")
```

Actually, I think we should be able to accept a state input (if we do it by typing) if it's lowercase. It's very easy to fix this with presentation logic, or in a controller.